docs/sources/tempo/api_docs/pushing-spans-with-http.md

```diff
@@ -9,101 +9,126 @@ aliases:
+        	"attributes": [{
+            	"key": "service.name",
+            	"value": {
+                	"stringValue": "my.service"
+            	}
+        	}]
+    	},
+    	"scopeSpans": [{
+        	"scope": {
+            	"name": "my.library",
+            	"version": "1.0.0",
+            	"attributes": [{
+                	"key": "my.scope.attribute",
+                	"value": {
+                    	"stringValue": "some scope attribute"
+                	}
+            	}]
+        	},
+        	"spans": [
+        	{
+            	"traceId": "5B8EFFF798038103D269B633813FC700",
+            	"spanId": "EEE19B7EC3C1B100",
+            	"name": "I am a span!",
+            	"startTimeUnixNano": 1689969302000000000,
+            	"endTimeUnixNano": 1689970000000000000,
+            	"kind": 2,
+            	"attributes": [
+            	{
+                	"key": "my.span.attr",
+                	"value": {
+                    	"stringValue": "some value"
+                	}
+            	}]
+        	}]
+    	}]
+	}]
+}'
 ```
 
-Note that the `timestamp` field is in microseconds and was obtained by running `date +%s%6N`.  The `duration` field is also in microseconds and so 100000 is 100 milliseconds.
+Note that the `startTimeUnixNano` field is in nanoseconds and can be obtained by any tool that provides the epoch date in nanoseconds (for example, under Linux, `date +%s%8N`). The `endTimeUnixNano` field is also in nanoseconds, where 100000000 nanoseconds is 100 milliseconds.
+
+1. Copy and paste the curl command into a text editor.
+
+1. Replace `startTimeUnixNano` and endTimeUnixNano with current values for the last 24 hours to allow you to search for them using a 24 hour relative time range. You can get this in seconds and milliseconds from the following link.
+Multiple the milliseconds value by 1,000,000 to turn it into nanoseconds. You can do this from a bash terminal with:
+
+   ```bash
+   echo $((<epochTimeMilliseconds> * 1000000))
+   ```
+
+1. Copy the updated curl command to a terminal window and run it.
+
+1. View the trace in Grafana:
+   1. Open a browser window to http://localhost:3000.
+   1. Open the Explorer page and select the Tempo data source.
+   1. Select the Search query type.
+   1. Select Run query to list available traces.
+   1. Select the trace ID (yellow box) to view details about the trace and its spans.
+
+![Using the TraceQL query builder on Explore to view pushed trace in Grafana.](/static/img/docs/tempo/push-spans-search-span-grafana.png "View the span in Grafana")
 
 ## Retrieve traces
 
 The easiest way to get the trace is to execute a simple curl command to Tempo.  The returned format is [OTLP](https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/trace/v1/trace.proto).
 
+1. Replace the trace ID in the curl command with the trace ID that was generated from the push (you could use Grafana’s Explorer page to find this, as shown in the previous section).
```

Comment:
It's also in the data that's sent with `curl`.

Reply:
I've added this to text. Thank you.